src/main/grammars/RustParser.bnf

```diff
@@ -314,9 +314,10 @@ fake ValueParameter ::= OuterAttr* Pat? TypeReference? {
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-FnParameter ::= !(Variadic) OuterAttr* [ Pat ':' ] TypeReference
-private FnParameter_with_recover ::= FnParameter (',' | &')') {
-  pin = 1
+FnParameter ::= OuterAttr* [ Pat ':' ] (VariadicUpper | TypeReference)
```

Comment:
Fixed double-parsing of `Pat` (a complex rule with a lot of `or`s). Upgrade to `Variadic` with `VariadicUpper`